src/Symfony/Component/HttpFoundation/ParameterBag.php

```diff
@@ -188,6 +188,35 @@ public function getBoolean($key, $default = false)
     }
 
     /**
+     * Returns the parameter value converted to a DateTime object.
+     *
+     * @param string             $key      The parameter key
+     * @param string             $format   The expected date format
+     * @param string             $default  The default value to be converted to a DateTime object if the parameter key does not exist
```

Comment:
I guess we can allow it for the `$default` parameter, although the combination of a `\DateTimeInterface` and a string is an odd one. 

I'm also a bit torn on the error handling aspect.

Usually I'd expect the function to throw an exception on malformed input. However returning fasly values is more PHP-esque(?). What's your input on this: @nicolas-grekas @xabbuh?